src/App.tsx

```diff
@@ -3,12 +1,80 @@
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { Filter } from './types/enumFilter';
 
 export const App: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [loader, setLoader] = useState(true);
+  const [flag, setFlag] = useState(Filter.All);
+  const [modalLoad, setModalLoad] = useState(true);
+
+  useEffect(() => {
+    getTodos()
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+      })
+      .finally(() => setLoader(false));
+  }, []);
+
+  const getFilteredByFlag = (filterFlag: Filter) => {
+    switch (filterFlag) {
+      case Filter.All:
+        setFlag(Filter.All);
+        getTodos().then(setTodos);
+        break;
+      case Filter.Active:
+        setFlag(Filter.Active);
+        getTodos()
+          .then(fitchedTodos => fitchedTodos.filter(t => !t.completed))
+          .then(setTodos);
+        break;
+      case Filter.Completed:
+        getTodos()
+          .then(fetchTodos => fetchTodos.filter(t => t.completed))
+          .then(setTodos);
+        break;
```

Comment:
In the `Filter.Completed` case, the `setFlag(Filter.Completed)` is missing. This should be added to ensure the state reflects the current filter.